### PR TITLE
[androiddebugbridge] Added mDNS discovery for Fire TV Stick

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added mDNS discovery for Fire TV Stick

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java [hunk 1]
```diff
@@ -370,17 +380,41 @@ public void checkConnection() {
...
+            AndroidDebugBridgeDeviceReadException, ExecutionException {
+        // Add some information about the device
+        try {
+            Map<String, String> editProperties = editProperties();
+            editProperties.put(Thing.PROPERTY_SERIAL_NUMBER, adbConnection.getSerialNo());
+            editProperties.put(Thing.PROPERTY_MODEL_ID, adbConnection.getModel());
+            editProperties.put(Thing.PROPERTY_FIRMWARE_VERSION, adbConnection.getAndroidVersion());
+            editProperties.put(Thing.PROPERTY_VENDOR, adbConnection.getBrand());
+            try {
+                editProperties.put(Thing.PROPERTY_MAC_ADDRESS, adbConnection.getMacAddress());
```
Comment (cweitkamp) [inline bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java:405]: Unfortunately the `ro.boot.wifimacaddr` property is not available or accessible on all devices but currently it looks like the Mac address is the only unique identifier available in both discovery service implementations.

File: bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/discovery/AndroidDebugBridgeDiscoveryService.java [hunk 2]
```diff
@@ -138,8 +142,15 @@ private void discoverWithADB(String ip, int port) throws InterruptedException, A
             String model = device.getModel();
             String androidVersion = device.getAndroidVersion();
             String brand = device.getBrand();
-            logger.debug("discovered: {} - {} - {} - {}", model, serialNo, androidVersion, brand);
-            onDiscoverResult(serialNo, ip, port, model, androidVersion, brand);
+            String macAddress;
+            try {
+                macAddress = device.getMacAddress();
```
Comment (cweitkamp) [inline bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/discovery/AndroidDebugBridgeDiscoveryService.java]: I think I can replace this call to retrieve the Mac address from the device by accessing the hardware address of the `NetworkInterface` used to discover the device.

File: bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java [hunk 3]
```diff
@@ -441,15 +446,13 @@ private String runAdbShell(String... args)
                 ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
                 String cmd = String.join(" ", args);
                 LOGGER.debug("{} - shell:{}", ip, cmd);
-                try {
-                    AdbStream stream = adb.open("shell:" + cmd);
+                try (AdbStream stream = adb.open("shell:" + cmd)) {
```
Comment (cweitkamp) [inline bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java:451]: Beside adding the mDNS discovery I think a found a potential memory leak here.

**Comment by J-N-K** [comment]
Will this also work on openHAB 3.1? Or at least not cause issues?

**Comment by cweitkamp** [comment]
Yes, I think but will work in OH 3.1 without any issues.

File: bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java [hunk 4]
```diff
@@ -370,17 +380,41 @@ public void checkConnection() {
...
+            } catch (AndroidDebugBridgeDeviceReadException e) {
+                // TODO: handle exception
```
Comment (J-N-K) [inline bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java]: Is this still a TODO?

Comment (cweitkamp) [reply bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java]: I solved it.

**Review by J-N-K — APPROVED** [review]

**Pull request merged**